File: frontend/src/types/decision.ts
export interface SalaryRecommendation {
  suggested: number | null;
  range: [number, number] | null;
  rationale: string;
}

export interface InterviewStageSummary {
  stage: string;
  fit_score: number;
  recommendation: string;
  summary: string;
}

export interface DecisionReport {
  overall_recommendation: 'hire' | 'reject' | 'hold';
  confidence: number;
  strengths_summary: string;
  risk_summary: string;
  technical_verdict: string;
  cultural_verdict: string;
  salary_recommendation: SalaryRecommendation;
  lessons_learned: string[];
  interview_stages_summary: InterviewStageSummary[];
}

export interface InterviewDecision {
  id: string;
  pipeline_id: string;
  job_id: string;
  candidate_id: string;
  final_decision: 'hire' | 'reject' | 'hold' | null;
  decision_reason: string | null;
  generated_report: DecisionReport | null;
  generation_model: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/models/decision.py
import uuid
from datetime import datetime

from sqlalchemy import String, Text, DateTime, JSON, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class InterviewDecision(Base):
    __tablename__ = "interview_decisions"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    pipeline_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("interview_pipeline.id", ondelete="CASCADE"), nullable=False
    )
    job_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("job_requisitions.id"), nullable=False
    )
    candidate_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("candidates.id"), nullable=False
    )
    final_decision: Mapped[str | None] = mapped_column(
        String(20)
    )  # hire | reject | hold
    decision_reason: Mapped[str | None] = mapped_column(Text)
    generated_report: Mapped[dict | None] = mapped_column(JSON)
    # {
    #   "overall_recommendation": "hire|reject|hold",
    #   "confidence": 0-100,
    #   "strengths_summary": "...",
    #   "risk_summary": "...",
    #   "technical_verdict": "...",
    #   "cultural_verdict": "...",
    #   "salary_recommendation": { "suggested": 85000, "range": [80000, 95000], "rationale": "..." },
    #   "lessons_learned": ["...", "..."],
    #   "interview_stages_summary": [{ "stage": "hr_interview", "fit_score": 78, "summary": "..." }]
    # }
    generation_model: Mapped[str | None] = mapped_column(String(100))
    primary_language: Mapped[str] = mapped_column(String(10), nullable=False, server_default="en")
    report_translations: Mapped[dict | None] = mapped_column(JSON)
    # JSON: { "zh": { "strengths_summary": "...", "risk_summary": "...", ... } }
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, default=datetime.utcnow, onupdate=datetime.utcnow
    )
